File: backend/erp-server/src/main/java/com/factory_dynamics/erp/erp_server/domain/ProductionPlan.java
package com.factory_dynamics.erp.erp_server.domain;

import jakarta.persistence.*;
import lombok.*;
import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "tb_production_plan")
public class ProductionPlan {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "plan_id")
    private Integer planId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "product_id", nullable = false)
    private Product product;

    @Column(name = "quantity", nullable = false, precision = 10, scale = 2)
    private BigDecimal quantity;

    @Column(name = "planned_start", nullable = false)
    private LocalDateTime plannedStart;

    @Column(name = "planned_end", nullable = false)
    private LocalDateTime plannedEnd;

    @Column(name = "status", nullable = false, length = 20)
    private String status;

    @Column(name = "is_deleted", columnDefinition = "boolean default false")
    private Boolean isDeleted = false;

    @Column(name = "deleted_at")
    private LocalDateTime deletedAt;

    @Column(name = "created_by", length = 50)
    private String createdBy;

    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @Column(name = "modified_by", length = 50)
    private String modifiedBy;

    @Column(name = "modified_at")
    private LocalDateTime modifiedAt;

    @PrePersist
    protected void onCreate() {
        this.createdAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        this.modifiedAt = LocalDateTime.now();
    }
}

File: frontend/ERP.Frontend/Models/ProductionPlan.cs
using System;
using Newtonsoft.Json;

namespace ERP.Frontend.Models
{
    public class ProductionPlan
    {
        [JsonProperty("planId")]
        public int PlanId { get; set; }

        [JsonProperty("product")]
        public Product Product { get; set; }

        [JsonProperty("quantity")]
        public decimal Quantity { get; set; }

        [JsonProperty("plannedStart")]
        public DateTime PlannedStart { get; set; }

        [JsonProperty("plannedEnd")]
        public DateTime PlannedEnd { get; set; }

        [JsonProperty("status")]
        public string Status { get; set; }

        // UI를 위한 추가 속성
        public string ProductName => Product?.Name;
    }
}